app/src/androidTest/java/ch/epfl/sdp/appart/AdCreationUITest.java

```diff
@@ -0,0 +1,151 @@
+package ch.epfl.sdp.appart;
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule;
+
+import androidx.test.espresso.intent.Intents;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+
+import ch.epfl.sdp.appart.database.DatabaseService;
+import ch.epfl.sdp.appart.database.MockDatabaseService;
+import ch.epfl.sdp.appart.hilt.DatabaseModule;
+import ch.epfl.sdp.appart.hilt.LoginModule;
+import ch.epfl.sdp.appart.login.LoginService;
+import ch.epfl.sdp.appart.login.MockLoginService;
+import dagger.hilt.android.testing.BindValue;
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
+import dagger.hilt.android.testing.UninstallModules;
+
+import static androidx.test.espresso.Espresso.closeSoftKeyboard;
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.scrollTo;
+import static androidx.test.espresso.action.ViewActions.typeText;
+import static androidx.test.espresso.assertion.ViewAssertions.matches;
+import static androidx.test.espresso.intent.Intents.intended;
+import static androidx.test.espresso.intent.matcher.IntentMatchers.hasComponent;
+import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+
+@UninstallModules({DatabaseModule.class, LoginModule.class})
+@HiltAndroidTest
+public class AdCreationUITest {
+
+    @Rule(order = 0)
+    public final HiltAndroidRule hiltRule = new HiltAndroidRule(this);
+
+    @Rule(order = 1)
+    public ActivityScenarioRule<AdCreationActivity> scrollingActivityRule = new ActivityScenarioRule<>(AdCreationActivity.class);
+
+    @Rule(order = 2)
+    public InstantTaskExecutorRule instantExecutorRule = new InstantTaskExecutorRule();
```

Comment:
What is the purpose of this line ?

Reply:
It serves to test livedata, the livedata we use executes the `postValue` function on a different thread which causes issues when testing since we need sequential execution. the instant executor makes sure everything is executed as if it happened in the same thread.

Reply:
Oh ok.